feat(.NET): Added obfuscation to troubleshooting (#12272)
getsentry/sentry-docs: docs/platforms/dotnet/common/troubleshooting.mdx

These issues have been fully resolved as of version 3.31.0 of the Sentry .NET SDK. To safely use an async main, update your version to the newest release.
It's no longer required to dispose of the result of calling `SentrySdk.Init`.

## Obfuscated Code and Stack Traces

> Sentry does not currently support de-obfuscation. This is because each obfuscator uses its own custom logic when dealing with deobfuscation, making it impractical to implement universal support.

Reviewer: ```suggestion

It's important to define what obfuscation means. If you're getting illegible stack traces because of missing [debug files](/platforms/dotnet/data-management/debug-files/), in order to fix it all you need to do is enabling debug file upload. That is done automatically by the [Sentry SDK via MSBuild](/platforms/dotnet/guides/aspnet/configuration/msbuild/).

If you're using a custom tool that obfuscates .NET Intermediate Language (IL) in order to make apps harder to reverse-engineer. Sentry does not currently support these tools. This is because each obfuscator uses its own custom logic when dealing with deobfuscation, making it impractical to implement universal support.
```

If you're working with obfuscated code, you have several options:

1. **Use a proxy service**: You can route exceptions through your own web service proxy where you can de-obfuscate the stack traces before sending them to Sentry. This gives you full control over the de-obfuscation process.

> 2. **Preserve method names**: Many obfuscation tools allow you to disable method name obfuscation while still obfuscating the code inside methods. This approach results in more readable stack traces while maintaining code protection. Configure your obfuscator to preserve method names if this balance of readability and security meets your needs.

Reviewer: ```suggestion
2. **Preserve method names**: Many obfuscation tools allow you to disable method name obfuscation while still obfuscating the code inside methods. This approach results in more readable stack traces while maintaining code protection. Configure your obfuscator to preserve method names if this balance of readability and security meets your needs.

3. **Contribute to Symbolic**: [Our symbolication library is open source](https://github.com/getsentry/symbolic). You could create a ticket to discuss adding support to it with the maintainers.
```

These issues have been fully resolved as of version 3.31.0 of the Sentry .NET SDK. To safely use an async main, update your version to the newest release.
It's no longer required to dispose of the result of calling `SentrySdk.Init`.

## Obfuscated Code and Stack Traces

> It's important to define what obfuscation means. If you're getting illegible stack traces because of missing [debug files](/platforms/dotnet/data-management/debug-files/), in order to fix it all you need to do is enabling debug file upload. That is done automatically by the [Sentry SDK via MSBuild](/platforms/dotnet/guides/aspnet/configuration/msbuild/).

Reviewer 2: ```suggestion
It's important to define what obfuscation means. If you're getting illegible stack traces because of missing [debug files](/platforms/dotnet/data-management/debug-files/), all you need to do is enabling debug file upload, which is done automatically by the [Sentry SDK via MSBuild](/platforms/dotnet/guides/aspnet/configuration/msbuild/).
```

> Some custom tools obfuscates .NET Intermediate Language (IL) in order to make apps harder to reverse-engineer; Sentry does not currently support these tools. This is because each obfuscator uses its own custom logic when dealing with deobfuscation, making it impractical to implement universal support.

Author: ```suggestion
Some custom tools obfuscate .NET Intermediate Language (IL) in order to make apps harder to reverse-engineer; Sentry does not currently support these tools. This is because each obfuscator uses its own custom logic when dealing with deobfuscation, making it impractical to implement universal support.
```

> If you're using a custom tool that obfuscates .NET Intermediate Language (IL) in order to make apps harder to reverse-engineer. Sentry does not currently support these tools. This is because each obfuscator uses its own custom logic when dealing with deobfuscation, making it impractical to implement universal support.

Reviewer 2: ```suggestion
Some custom tools obfuscates .NET Intermediate Language (IL) in order to make apps harder to reverse-engineer; Sentry does not currently support these tools. This is because each obfuscator uses its own custom logic when dealing with deobfuscation, making it impractical to implement universal support.
```

If you're working with obfuscated code, you have several options:

> 1. **Use a proxy service**: You can route exceptions through your own web service proxy where you can de-obfuscate the stack traces before sending them to Sentry. This gives you full control over the de-obfuscation process.

Reviewer 2: ```suggestion
1. **Use a proxy service**: You can route exceptions through your own web service proxy where you can deobfuscate the stack traces before sending them to Sentry. This gives you full control over the deobfuscation process.
```
